### PR TITLE
Try to wait before sending more than 60 messages/min and retry sends on error

### PR DESCRIPTION
**Pull request by technillogue**
yay

* sketch for retrying on rate limit
* have write_commands also sleep when retrying after 413
* logging
* enqueue errors to be processed so 409s can be matched to messages. add a base rate limit on sending messages of 1msg/sec, cap 50
* reword names

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201632126650504/1201662871755183) by [Unito](https://www.unito.io)

### CODE REVIEW

File: forest/core.py [hunk 1]
```diff
@@ -750,30 +797,41 @@ async def build_gift_code(self, amount_pmob: int) -> list[str]:
...
+        tx_id = raw_prop["result"]["transaction_log_id"]
+        if confirm_tx_timeout:
+            await self.mob_request(
+                "submit_transaction",
+                tx_proposal=prop,
+                comment=comment,
+                account_id=account_id,
+            )
+        else:
+            await self.mob_request("submit_transaction", tx_proposal=prop)
```
Comment (technillogue) [inline forest/core.py:834]: @iamalwaysuncomfortable why is this if/else necessary? should the unconfirmed transactions not have account_id/comment?

File: forest/core.py [hunk 2]
```diff
@@ -785,9 +843,49 @@ async def send_payment(
...
+                await asyncio.sleep(1)
+
+            if status == "tx_status_pending":
+                logging.warning(
+                    "Tx to %s timed out - tx data: %s",
+                    recipient,
+                    tx_status.get("result"),
+                )
+            resp = await resp_fut
+            resp.status, resp.transaction_log_id = status, tx_id  # type: ignore
```
Comment (technillogue) [inline forest/core.py:883]: @iamalwaysuncomfortable what's going on here?

**Comment by iamalwaysuncomfortable** [comment]
The if else can be slight restructured, but the goal roughly is to have the ability to NOT log transactions into the full service DB if the sender wants it private. Putting the account_id into the request logs it to full service, if you omit it, it doesn't get logged. The if else is basically to give the bot author options to not log it if they don't want.

File: forest/core.py (hunk 2 above)
Comment (technillogue) [reply forest/core.py:889]: bumping this, if I'm following correctly resp is a message with e.g. `.blob = {"timestamp":1642647822493}}`. Where is  transaction_log_id used?

**Comment by technillogue** [comment]
> The if else can be slight restructured, but the goal roughly is to have the ability to NOT log transactions into the full service DB if the sender wants it private. Putting the account_id into the request logs it to full service, if you omit it, it doesn't get logged. The if else is basically to give the bot author options to not log it if they don't want.

cool, sounds good - can you add a comment with that?

**Pull request merged**